From 1c2639a5468e11909c9cebe0f3a3dc7e13ef3811 Mon Sep 17 00:00:00 2001
From: Michael Tokarev <mjt@tls.msk.ru>
Date: Fri, 20 May 2022 09:48:32 +0300
Subject: [PATCH] testparm: clarify "Weak crypto is allowed" message

The message testparm prints about weak crypto is really
misleading: "Weak crypto is allowed" is often interpreted
in a way that smb.conf settings are bad by allowing weak
crypto.  While the actual meaning is about the ability to
fall back to weaker crypto for (backwards) compatibility,
and this has nothing to do with samba settings, it is the
gnutls settings. Clarify both of these, and eliminate an
if() and a local variable.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>
---
 source3/utils/testparm.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/source3/utils/testparm.c b/source3/utils/testparm.c
index 71bc4c2694e..b32ba05322c 100644
--- a/source3/utils/testparm.c
+++ b/source3/utils/testparm.c
@@ -735,7 +735,6 @@ static void do_per_share_checks(int s)
 	const char *caddr;
 	static int show_defaults;
 	static int skip_logic_checks = 0;
-	const char *weak_crypo_str = "";
 	bool ok;
 
 	struct poptOption long_options[] = {
@@ -870,12 +869,8 @@ static void do_per_share_checks(int s)
 
 	fprintf(stderr,"Loaded services file OK.\n");
 
-	if (samba_gnutls_weak_crypto_allowed()) {
-		weak_crypo_str = "allowed";
-	} else {
-		weak_crypo_str = "disallowed";
-	}
-	fprintf(stderr, "Weak crypto is %s\n", weak_crypo_str);
+	fprintf(stderr, "Weak crypto is %sallowed (compatibility fallback; gnutls setting)\n",
+	        samba_gnutls_weak_crypto_allowed() ? "" : "dis");
 
 	if (skip_logic_checks == 0) {
 		ret = do_global_checks();
-- 
2.30.2

